### PR TITLE
feat: improve exchange error handler

### PR DESCRIPTION
**Pull request by mckrava**

### CODE REVIEW

File: src/methods/tx/_events.ts [hunk 1]
```diff
@@ -241,6 +245,23 @@ export const processChainEvent = (
          *                     [who, assets, sell or buy, intention id, error detail]
          */
         if (Array.isArray(parsedData)) {
+          const dispatchError = parsedData[4];
+          let errorDetails = dispatchError.toString();
+
+          if (parsedData[4].module !== undefined) {
+            const { documentation, section, name } = api.registry.findMetaError(
+              {
+                error: new BN(dispatchError.module.error),
```
Comment (smupyknight) [inline src/methods/tx/_events.ts:254]: Do we have guarantee that module.error is always number? What happens if it is string value?

**Comment by smupyknight** [comment]
I just have one comment on it. Other than that, looks fine.

File: src/methods/tx/_events.ts (hunk 1 above)
Comment (mckrava) [reply src/methods/tx/_events.ts:258]: I've added value check so now value must be a number.

**Comment by smupyknight** [comment]
Looks fine.

**Pull request merged**